tests/meson.build

```diff
@@ -3,47 +3,28 @@
 # If feature 'unit_tests' is set to 'auto', then unit tests are disabled
 # for release builds and enabled everywhere else.
 #
-# Meson will first try to find pre-installed gtest via pkg-config;
+# Meson will first try to find pre-installed gmock and gtest via pkg-config;
 # if this fails, it will download wrap dependency (subproject).
 #
 # Users can further configure the download behaviour using:
 #
 #   meson setup --wrap-mode={nofallback,nodownload,forcefallback}
 #
-# If gtest is not available at all, unit tests will be disabled.
+# If gmock and gtest are not available at all, unit tests will be disabled.
 #
-gtest_dep = optional_dep
 gmock_dep = optional_dep
 
 if not (get_option('buildtype') == 'release' and get_option('unit_tests').auto())
-  gtest_system_dep = dependency('gtest', main : true,
-                                required : get_option('unit_tests'))
-  gmock_system_dep = dependency('gmock', main : true,
-                                required : get_option('unit_tests'))
-
-  # GTest and GMock are closely coupled together, we must use them in exactly
-  # the same version.  We can't allow situation when one of packages is
-  # system-installed and another one is used from meson subproject.
-  #
-  if (gmock_system_dep.found() and gtest_system_dep.found())
-    gtest_dep = gtest_system_dep
-    gmock_dep = gmock_system_dep
-  else
-    message('GTest or GMock is not installed, using fallback for both dependencies')
-    gtest_dep = dependency('', main : true,
-                           required : get_option('unit_tests'),
-                           fallback : ['gtest', 'gtest_main_dep'])
-    gmock_dep = dependency('', main : true,
-                           required : get_option('unit_tests'),
-                           fallback : ['gtest', 'gmock_dep'])
-  endif
+  gmock_dep = dependency('gmock', main : true,
+                         version : '> 1.8.0',
+                         required : get_option('unit_tests'),
+                         fallback : ['gtest', 'gmock_main_dep'])
 endif
-if not (gtest_dep.found() and gmock_dep.found())
-  gtest_dep = disabler()
+if not gmock_dep.found()
   gmock_dep = disabler()
 endif
```

Comment:
Since Meson 0.49.0 (so well within your existing requirements) you can just pass the `disabler: true` kwarg to dependency() and it will automatically be a disabler if it is not found.

Reply:
`meson setup --wrap-mode=nofallback -Duse_fluidsynth=false -Duse_slirp=false build` (without system gmock) -> Unit Test disabled.

Works like a charm, thanks! :)